hazelcast/src/main/java/com/hazelcast/spi/impl/eventservice/impl/RegistrationUtil.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.spi.impl.eventservice.impl;
+
+import javax.annotation.Nonnull;
+import java.util.UUID;
+import java.util.concurrent.Future;
+
+import static com.hazelcast.internal.util.ExceptionUtil.rethrow;
+
+public interface RegistrationUtil {
```

Comment:
This two methods are doing the same thing. I would merge them into one function with generics and also put it under FutureUtil instead of here. Also InterruptedException should be better handled. 
```
 @Nonnull
    static <T> T join(Future<T> future) {
        try {
            return future.get();
        } catch (InterruptedException e) {
            Thread.currentThread().interrupt();
            throw rethrow(e);
        } catch (ExecutionException e) {
            throw rethrow(e);
        }
    }
```